ts/parts/Axis.ts

```diff
@@ -7434,7 +7443,32 @@ extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
                         });
                     }
                 }
+            }
+
+            // Clip crosshair of the circular axis only when the pane's
+            // innerSize is bigger than 0
+            if (this.isCircular && center && center[3] > 0) {
+                clipCrosshair = this.clipCrosshair;
+                clipOptions = {
+                    x: center[0] + chart.plotLeft,
+                    y: center[1] + chart.plotTop,
+                    r: center[2] / 2,
+                    innerR: center[3] / 2
+                };
 
+                // Create or update clip
+                this.clipCrosshair = clipCrosshair ?
+                    clipCrosshair.attr(clipOptions) :
+                    chart.renderer.clipCircle(
+                        clipOptions.x,
+                        clipOptions.y,
+                        clipOptions.r,
+                        clipOptions.innerR
+                    );
+
+                if (this.cross) {
+                    this.cross.clip(this.clipCrosshair);
+                }
```

Comment:
This part should be refactored out so we don't pollute the basic `Axis` with polar-specific stuff. We could possibly do it in an event call, `afterDrawCrosshairs`, but I'm wondering if clipping is necessary at all. The crosshair path is calculated using `getPlotLinePath` further up, and I would expect this to be a correct path within the pane`s `size` and `innerSize` bounds.

Reply:
Yes, this could probably be done inside the `getPlotLinePath` (in the section where `a` and `b` properties are calculated to find an adequate ratio). I'll check that and correct it. Thanks!

Reply:
It seems that I have already provided such logic for grid lines and somehow I missed that it also works for circular `crosshair` therefore clipping (which I have added before) was unnecessary.